### PR TITLE
 DRIVERS-828 add tests for uncaptured errors and failures

### PR DESCRIPTION
**Pull request by p-mongo**
Test that uncaptured errors and failures fail the test run.

This PR also adds sleeps on each iteration of validator scenarios to reduce the amount of logs produced and speed up validation (since astrolabe takes some time to parse the produced json files).

### CODE REVIEW

File: astrolabe/utils.py [hunk 1]
```diff
@@ -266,7 +268,41 @@ def terminate(self):
...
             sleep(2)
+            
+        return self.read_stats()
+        
+    def terminate(self):
+        '''Ensure the process is terminated.'''
+        
+        LOGGER.info("Stopping workload executor [PID: {}]".format(self.pid))
+
+        try:
```
Comment (prashantmital) [inline astrolabe/utils.py:279]: This try...except seems to be the only difference between the functions... can you explain what we can't just have one function for this?

Comment (p-mongo) [reply astrolabe/utils.py:251]: Both functions combined into one per our discussion.

File: astrolabe/validator.py [hunk 2]
```diff
@@ -213,10 +275,10 @@ def test_num_failures_as_errors(self):
 
         num_reported_errors = stats['numErrors']
         num_reported_failures = stats['numFailures']
-        if abs(num_reported_errors - num_reported_finds) > 1:
+        if abs(num_reported_errors - num_reported_finds/2) > 1:
```
Comment (prashantmital) [inline astrolabe/validator.py:278]: Where do the `/2` come from?

File: astrolabe/utils.py (hunk 1 above)
Comment (prashantmital) [reply astrolabe/utils.py:251]: 👍🏽

File: astrolabe/utils.py [hunk 3]
```diff
@@ -243,14 +243,19 @@ def spawn(self, *, workload_executor, connection_string, driver_workload,
...
-        else:
-            os.kill(self.workload_subprocess.pid, signal.CTRL_BREAK_EVENT)
-
+        
+        try:
+            if not self.is_windows:
+                os.killpg(self.workload_subprocess.pid, signal.SIGINT)
+            else:
+                os.kill(self.workload_subprocess.pid, signal.CTRL_BREAK_EVENT)
+        except ProcessLookupError as exc:
```
Comment (prashantmital) [inline astrolabe/utils.py:256]: Nice! 🎉

**Review by prashantmital — COMMENTED** [review]
Everything looks good but I don't understand what introduced all the divide-by-2 logic? Can you explain that to me?

**Review by prashantmital — APPROVED** [review]

**Pull request merged**